src/vec.rs

```diff
@@ -336,6 +348,7 @@ where
 ///
 /// [`Vec`]: (https://doc.rust-lang.org/std/vec/struct.Vec.html)
 ///
+#[derive(Clone)]
```

Comment:
This should be a manual `Clone` implementation. No need for a `Clone` bound on `N`.